packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -718,6 +718,15 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
         this.channel.reSubmit(innerContents.type, innerContents.content, localOpMetadata);
     }
 
+    public rollback(contents: any, localOpMetadata: unknown) {
+        assert(!!this.channel, "Channel must exist when rolling back ops");
```

Comment:
Can this be violated by using the APIs in a certain way, or we believe our code guarantees this invariant?

Reply:
The way the full flow works, it shouldn't happen. But I guess it's possible for new code paths to do something different.